### PR TITLE
IGNITE-6324 Transactional changes partially available after recovery.

### PR DESCRIPTION
**Pull request by zstan**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxLocalAdapter.java [hunk 1]
```diff
@@ -937,7 +938,15 @@ assert ownsLock(txEntry.cached()) :
                     U.warn(log, "Failed to commit transaction, node is stopping [tx=" + CU.txString(this) +
                         ", err=" + ex + ']');
 
-                    return;
+                    boolean persistenceEnabled = CU.isPersistenceEnabled(cctx.kernalContext().config());
+
+                    if (persistenceEnabled) {
+                        GridCacheDatabaseSharedManager dbManager = (GridCacheDatabaseSharedManager)cctx.database();
+
+                        dbManager.skipCheckPointOnNodeStop(true);
```
Comment (sergey-chugunov-1985) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxLocalAdapter.java:946]: Why not to replace this call with something like `dbManager.getCheckpointer().skipCheckpointOnNodeStop(true)`?

By this we'll eliminate two unnecessary methods on dbManager and checkpointManager.

What do you think?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java [hunk 2]
```diff
@@ -190,6 +193,10 @@
 import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.CORRUPTED_DATA_FILES_MNTC_TASK_NAME;
 import static org.apache.ignite.internal.util.IgniteUtils.GB;
 import static org.apache.ignite.internal.util.IgniteUtils.checkpointBufferSize;
+import static org.apache.ignite.transactions.TransactionState.COMMITTED;
```
Comment (sergey-chugunov-1985) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java:196]: I don't like the fact the we add a lot of details about transactions into DatabaseSharedManager which be aware of this stuff as little as possible.

Why not to introduce a filter (e.g. we may call it `NonFinishedTxFilter`) and put it into IgniteTxManager. All the structures like `uncommitedTx` set and logic will go there, database manager will just notify the filter about TX records and ask whether next transaction should be skipped.

What do you think about that?

**Review by sergey-chugunov-1985 — CHANGES_REQUESTED** [review]
@zstan, PR in general looks logical and simple enough for me, but I left two comments about code structure. Could you please take a look?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java (hunk 2 above)
Comment (zstan) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java:196]: huge +1 ! thanks for pointing this !

**Pull request closed without merging**